modules/ai/connect/ai.php

```diff
@@ -398,6 +400,74 @@ public function get_image_to_image( $image_data, $context, $request_ids ) {
 		return $result;
 	}
 
+
+	private function resizeImageIfNeeded( $original_url ) {
+		try {
+			$max_file_size = 4194304;
+			$current_size = filesize( $original_url );
+
+			if ( $current_size <= $max_file_size ) {
+				return $original_url;
+			}
+
+			$image_editor = wp_get_image_editor( $original_url );
+
+			if ( is_wp_error( $image_editor ) ) {
+				return $original_url;
+			}
+
+			$dimensions = $image_editor->get_size();
+			$original_width = $dimensions['width'];
+			$original_height = $dimensions['height'];
+
+			$scaling_factor = sqrt( $max_file_size / $current_size );
+
+			$new_width = (int) ( $original_width * $scaling_factor );
+			$new_height = (int) ( $original_height * $scaling_factor );
+
+			$image_editor->resize( $new_width, $new_height, true );
+
+			$file_extension = pathinfo( $original_url, PATHINFO_EXTENSION );
+			$temp_image = tempnam( sys_get_temp_dir(), 'resized_' ) . '.' . $file_extension;
+
+			$image_editor->save( $temp_image );
```

Comment:
Who is cleaning the temp files?

Reply:
the one who use it, in this case - get_unify_product_images